src/relation_engine_client/main.py

```diff
@@ -0,0 +1,151 @@
+import os
+import json
+import requests
+from typing import Optional, List, Dict, Union
+from dataclasses import dataclass
+
+from src.relation_engine_client.exceptions import REServerError, RERequestError, RENotFound
+
+_QUERY_METHOD = 'POST'
+_QUERY_ENDPOINT = '/api/v1/query_results'
+_SAVE_METHOD = 'PUT'
+_SAVE_ENDPOINT = '/api/v1/documents'
+
+
+@dataclass
+class REClient:
+    # The `api_url` can be set with the RE_API_URL env var if provided.
+    # We can also use the KBASE_ENDPOINT env var (eg. "https://ci.kbase.us/services/").
+    api_url: Optional[str] = None
+    # Set to the KBASE_TOKEN env var if not provided
+    token: Optional[str] = None
+
+    def __post_init__(self):
+        if self.token is None:
+            self.token = os.environ.get('KBASE_TOKEN')
```

Comment:
I really don't like looking up stuff in the environment in library code - it's too magic and leads to unexpected behavior (for instance in the narrative it's a real pain to get an anonymous client). I'd prefer helper methods to get the values from the environment so the client programmer can choose to use the environmental values or not.